compress/decompress.go

```diff
@@ -712,32 +735,47 @@ func (g *Getter) Match(buf []byte) (bool, uint64) {
 		bufPos += int(pos) - 1
 		if bufPos > lastUncovered {
 			dif := uint64(bufPos - lastUncovered)
-			if lenBuf < bufPos || !bytes.Equal(buf[lastUncovered:bufPos], g.data[postLoopPos:postLoopPos+dif]) {
+			compared := bytes.Compare(buf[lastUncovered:bufPos], g.data[postLoopPos:postLoopPos+dif])
+			if compared != 0 {
 				g.dataP, g.dataBit = savePos, 0
-				return false, savePos
+				return compared
+			}
+			if lenBuf < bufPos {
+				g.dataP, g.dataBit = savePos, 0
+				return -1
 			}
 			postLoopPos += dif
 		}
 		lastUncovered = bufPos + len(g.nextPattern())
 	}
 	if int(wordLen) > lastUncovered {
 		dif := wordLen - uint64(lastUncovered)
-		if lenBuf < int(wordLen) || !bytes.Equal(buf[lastUncovered:wordLen], g.data[postLoopPos:postLoopPos+dif]) {
+
+		compared := bytes.Compare(buf[lastUncovered:wordLen], g.data[postLoopPos:postLoopPos+dif])
+		if compared != 0 {
+			g.dataP, g.dataBit = savePos, 0
+			return compared
+		}
+		if lenBuf < int(wordLen) {
 			g.dataP, g.dataBit = savePos, 0
-			return false, savePos
+			return -1
 		}
 		postLoopPos += dif
 	}
-	if lenBuf != int(wordLen) {
+	if lenBuf < int(wordLen) {
 		g.dataP, g.dataBit = savePos, 0
-		return false, savePos
+		return -1
+	}
+	if lenBuf > int(wordLen) {
+		g.dataP, g.dataBit = savePos, 0
+		return 1
 	}
 	g.dataP, g.dataBit = postLoopPos, 0
-	return true, postLoopPos
+	return 0
 }
 
 // MatchPrefix only checks if the word at the current offset has a buf prefix. Does not move offset to the next word.
-func (g *Getter) MatchPrefix(prefix []byte) bool {
+func (g *Getter) MatchPrefix(prefix []byte, nf uint64) (cmp int, fullMatchOfKey bool, nextOffset uint64) {
```

Comment:
What is “nf”?

Reply:
This will be removed. I was testing that the next offset is correct, so I had to pass returned value again to function.